pkg/controllers/state/cluster.go

```diff
@@ -461,18 +465,27 @@ func (c *Cluster) cleanupNode(name string) {
 }
 
 func (c *Cluster) populateStartupTaints(ctx context.Context, n *StateNode) error {
+	if n.startupTaintsInitialized {
```

Comment:
nit: just a comment here        
```suggestion
        // We only need to populate the startup taints once.
	if n.startupTaintsInitialized {
```